runtime/stdlib/cadence_v0.42_to_v1_contract_upgrade_validator.go

```diff
@@ -46,11 +47,20 @@ func NewCadenceV042ToV1ContractUpdateValidator(
 	contractName string,
 	provider AccountContractNamesProvider,
 	oldProgram *ast.Program,
-	newProgram *ast.Program,
+	newProgram *interpreter.Program,
```

Comment:
Accept an `interpreter.Program`, because we also need the elaboration of the current program